test/integration/invalid-packet-stream-test.js

```diff
@@ -0,0 +1,49 @@
+const { assert } = require('chai');
+const net = require('net');
+const Connection = require('../../src/tedious').Connection;
+const ConnectionError = require('../../src/errors').ConnectionError;
+
+describe('Connecting to a server that sends invalid packet data', function() {
+    let server;
+  
+    beforeEach(function(done) {
+      server = net.createServer();
+      server.listen(0, '127.0.0.1', done);
+    });
+  
+    afterEach(function(done) {
+      server.close(done);
+      server.emit('close')
+    });
+  
+    it('should throw Connection Error ', function(done) {
+      server.on('connection', (socket) => {
+        const packetData = Buffer.from('test1234');
+        const packetHeader = Buffer.alloc(8);
+        let offset = 0;
+        offset = packetHeader.writeUInt8(0x11, offset);
+        offset = packetHeader.writeUInt8(0x01, offset);
+        offset = packetHeader.writeUInt16BE(5, offset);
+        offset = packetHeader.writeUInt16BE(0x0000, offset);
+        offset = packetHeader.writeUInt8(1, offset);
+        packetHeader.writeUInt8(0x00, offset);
+        const packet = Buffer.concat([packetHeader, packetData]);
+        socket.write(packet);
+      });
+      const addressInfo = server.address();
+      const connection = new Connection({
+        server: addressInfo.address,
+        options: {
+          port: addressInfo.port,
+        }
+      });
+      connection.connect((err) => {});
+  
+      connection.on('error', (err) => {
+        assert.instanceOf(err, ConnectionError);
+        assert.equal(err.message, 'Unable to process incoming packet');
+        connection.close();
+        done();
+      })
```

Comment:
Something is not quite right here. 🤔 

There should be an `err` being passed to the callback given to `connect`, but no error emitted via the  `error` event. We currently only emit `error` events if the connection was established successfully, and I think we have to continue doing that.

Maybe instead of doing

```ts
        this.emit('error', error);
        this.dispatchEvent('socketError', error);
```

we should be just calling `socketError`, like this:

```ts
        this.socketError(error);
```

Can you check if that would change the behavior to what we want? 🙇‍♂️

Reply:
`connection.connect(cb)` calls the callback when the first `connect` event is emitted. So `this.emit('connect', error)` seems to do the trick.